test/common/secret/sds_api_test.cc

```diff
@@ -105,13 +105,14 @@ TEST_F(SdsApiTest, DynamicCertificateValidationContextUpdateSuccess) {
   NiceMock<Server::MockInstance> server;
   NiceMock<Init::MockManager> init_manager;
   envoy::api::v2::core::ConfigSource config_source;
-  CertificateValidationContextSdsApi sds_api(
-      server.localInfo(), server.dispatcher(), server.random(), server.stats(),
-      server.clusterManager(), init_manager, config_source, "abc.com", []() {});
+  auto sds_api = std::static_pointer_cast<CertificateValidationContextSdsApi>(
+      CertificateValidationContextSdsApi::create(
+          server.localInfo(), server.dispatcher(), server.random(), server.stats(),
+          server.clusterManager(), init_manager, config_source, "abc.com", []() {}));
```

Comment:
you don't need to change this.  It can still use constructor

Reply:
Reverted.